96.md

```diff
@@ -0,0 +1,192 @@
+NIP-96
+======
+
+HTTP File Storage Integration
+-----------------------------
+
+`draft` `optional` `author:arthurfranca` `author:Semisol` `author:staab`
+
+## Introduction
+
+This NIP defines a REST API for HTTP file storage servers intended to be used in conjunction with the nostr network.
+The API will enable nostr users to upload files and later reference them by url on nostr notes.
+
+The spec DOES NOT use regular nostr events through websockets for
+storing, requesting nor retrieving data because, for simplicity, the server
+will not have to learn anything about nostr relays.
+
+## Server Adaptation
+
+File storage servers wishing to be accessible by nostr users should opt-in by making available
+an https route at `/.well-known/nostr.json` with `nip96.routes` field in the following form:
+
+```js
+{
+  "nip96": {
+    // required
+    "routes": {
+      "upload": "https://upload.domain/upload-path",
+      "download": "https://download.domain/download-path",
+      "deletion": "https://deletion.domain/deletion-path"
+    },
+    // optional
+    "nip_integrations": [],
+    // optional
+    "tos_url": "https://my.landing.page/terms-of-service",
+    // optional
+    "plans": {
+      // "free" is the only standardized plan key and
+      // clients may use its presence to learn if server offers free storage
+      "free": {
+        "name": "Free Tier",
+        "url": "https://...", // payment etc
+        "max_byte_size": 10485760,
+        // range in days / 0 for no expiration
+        // [7, 0] means it may vary from 7 days to unlimited persistence,
+        // [0, 0] means it has no expiration
+        // early expiration may be due to low traffic or any other factor
+        "file_expiration": [14, 90],
+        "image_transformations": ["resizing"],
+        "cost": {
+          // it considers lowest currency unit
+          // "100" USD equals 100 cents or 1 dollar
+          // "1000" BTC equals 1000 milisats or 1 satoshi
+          "price": "0", // string
+          "currency": "USD",
+          // "per-upload|one-time|day|week|month|year..."
+          "frequency": ""
+        }
+      }
+    }
+  }
+}
```

Comment:
Yes, you version (root_url) is in fact very similar to how it was written originally. But because [of this](https://github.com/nostr-protocol/nips/pull/547#discussion_r1256083467) I had to change it.

Reply:
People complained of reserving paths such as /file. Also PUT isn't supported by html form fields that's why I used POST.

Ended up with api_url (can include custom paths) and optional download_url.